pkgs/development/compilers/llvm/common/libunwind/default.nix

```diff
@@ -109,9 +109,13 @@ stdenv.mkDerivation (
       + lib.optionalString (enableShared && stdenv.hostPlatform.isWindows) ''
         ln -s $out/lib/libunwind.dll.a $out/lib/libunwind_shared.dll.a
       ''
+      + lib.optionalString (doFakeLibgcc && !stdenv.hostPlatform.isWindows && enableShared) ''
+        ln -s $out/lib/libunwind${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libgcc_s${stdenv.hostPlatform.extensions.sharedLibrary}
+        ln -s $out/lib/libunwind${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libgcc_s${stdenv.hostPlatform.extensions.sharedLibrary}.1
+      ''
       + lib.optionalString (doFakeLibgcc && !stdenv.hostPlatform.isWindows) ''
```

Comment:
Is the static library always built even when `enableShared`?

Note that this is inconsistent with GCC’s `libgcc` output, which only contains `libgcc_s.so`, and `doFakeLibgcc` in `compiler-rt`, which installs a `libgcc.a` from `libclang_rt.builtins-*.a`. I am not sure this produces correct results (well, as correct as `doFakeLibgcc` is already, even).

Reply:
Yes, the static library always exists

Reply:
> Isn’t this just going to produce clashing libraries in the environment that win based on the dependency order? That sounds like it’s more likely to break things than fix them?

Probably but it was already doing that.

> If it’s not suitable as a `libgcc` by itself, we shouldn’t install it as one. The derivation that creates something that works as a `libgcc` can make it out of `libunwind`.

I'm not sure in terms of compatibility what the differences are.

> Aren’t we just reinventing `llvm-libgcc`, which already has the relevant linker script incantations?

Yes but at least this works when `llvm-libgcc` doesn't when I've tried packaging it.

Reply:
> Probably but it was already doing that.

I don’t think so? `compiler-rt` produces a `libgcc.a` and `libunwind` produces a `libgcc_s.so`. Those are different things. This PR would make them clash.

> Yes but at least this works when `llvm-libgcc` doesn't when I've tried packaging it.

It’d probably be easier to figure out what the issues are if you could put up a draft PR for that work. At least, creating intermediate artefacts with the same name as the built product seems confusing and like it will only create issues, if the plan is to combine them in another derivation, even if we’re diverging from upstream on that part.

Reply:
#443065